src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -32,9 +31,12 @@ internal class AzureAppConfigurationProvider : ConfigurationProvider, IConfigura
         private Uri _lastSuccessfulEndpoint;
         private AzureAppConfigurationOptions _options;
         private Dictionary<string, ConfigurationSetting> _mappedData;
-        private Dictionary<KeyValueIdentifier, ConfigurationSetting> _watchedSettings = new Dictionary<KeyValueIdentifier, ConfigurationSetting>();
+        private Dictionary<KeyValueIdentifier, ConfigurationSetting> _watchedIndividualKvs = new Dictionary<KeyValueIdentifier, ConfigurationSetting>();
+        private Dictionary<KeyValueSelector, IEnumerable<MatchConditions>> _watchedKvEtags = new Dictionary<KeyValueSelector, IEnumerable<MatchConditions>>();
```

Comment:
It doesn't look like watched etags need to be kept separate given key value selector gives a way to distinguish.

Reply:
Yea good point, I think making it 1 dictionary could make things simpler even though we have to filter out the kv/ff etags each time. I can try that

Reply:
not pushed in latest commit, will just make sure it all works first